test/components/admin/editions/tags_component_test.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Admin::Editions::TagsComponentTest < ViewComponent::TestCase
+  [
+    {
+      state: :force_published,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--yellow",
+      label_text: "Force published",
+    },
+    {
+      state: :draft,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--blue",
+      label_text: "Draft",
+    },
+    {
+      state: :submitted,
+      expected_tag_classes: "govuk-tag govuk-tag--s",
+      label_text: "Submitted",
+    },
+    {
+      state: :published,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--green",
+      label_text: "Published",
+    },
+    {
+      state: :scheduled,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--turquoise",
+      label_text: "Scheduled",
+    },
+    {
+      state: :rejected,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--red",
+      label_text: "Rejected",
+    },
+    {
+      state: :withdrawn,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--grey",
+      label_text: "Withdrawn",
+    },
+    {
+      state: :unpublished,
+      expected_tag_classes: "govuk-tag govuk-tag--s govuk-tag--grey",
+      label_text: "Unpublished",
+    },
+  ].each do |hash|
+    test "returns the correct tag for the #{hash[:state]} state" do
+      state = hash[:state]
+      edition = state == :unpublished ? create(:edition, state) : build(:edition, state)
```

Comment:
Would it be simpler just to `create` them all here?

```suggestion
      edition = create(:edition, hash[:state])
```

I imagine the `unpublished` state needs the Edition to exist in the database (i.e. with `create`) so the separate Unpublishing record can be created and attached to it. But that seems like a detail which probably isn't helpful context for a developer to need when working in this area of code.

Reply:
yea that's exactly right, i just did this to speed the tests up but happy to use create.